File: 1662-check-if-two-string-arrays-are-equivalent/1662-check-if-two-string-arrays-are-equivalent.cpp
class Solution {
public:
    bool arrayStringsAreEqual(vector<string>& word1, vector<string>& word2) {
        int n1 = word1.size();
        int n2 = word2.size();
        int i = 0;
        int j = 0;
        int leni = word1[0].length();
        int lenj = word2[0].length();
        int ki = 0;
        int kj = 0;
        while ((i < n1) & (j < n2)) {
            if (word1[i][ki] != word2[j][kj]) {
                return false;
            }
            ki++;
            if (ki == leni) {
                ki = 0;
                i++;
                if (i < n1) {
                    leni = word1[i].length();
                }
            }
            kj++;
            if (kj == lenj) {
                kj = 0;
                j++;
                if (j < n2) {
                    lenj = word2[j].length();
                }
            }
        }
        return ((i == n1) & (j == n2));
    }
};

File: 1662-check-if-two-string-arrays-are-equivalent/1662-check-if-two-string-arrays-are-equivalent.py
class Solution:
    def arrayStringsAreEqual(self, word1: List[str], word2: List[str]) -> bool:
        if reduce(lambda x, y: x + y, map(lambda x: len(x), word1)) != reduce(lambda x, y: x + y, map(lambda x: len(x), word2)):
            return False
        return "".join(word1) == "".join(word2)
    """
    def arrayStringsAreEqual(self, word1: List[str], word2: List[str]) -> bool:
        n1 = len(word1)
        n2 = len(word2)
        i = 0
        j = 0
        leni = len(word1[0])
        lenj = len(word2[0])
        ki = 0
        kj = 0
        while (i < n1) and (j < n2):
            if word1[i][ki] != word2[j][kj]:
                return False
            ki += 1
            if ki == leni:
                ki = 0
                i += 1
                if i < n1:
                    leni = len(word1[i])
            kj += 1
            if kj == lenj:
                kj = 0
                j += 1
                if j < n2:
                    lenj = len(word2[j])
        return (i == n1) and (j == n2)
    """
